span/src/lib.rs

```diff
@@ -14,8 +14,8 @@
 mod line;
 mod span;
 
-pub use line::Line;
-pub use span::Span;
+pub use crate::line::Line;
+pub use crate::span::Span;
```

Comment:
I don't understand this patch. Where is the ambiguity? We clearly defined the modules in the previous lines.

Reply:
I started seeing this fail in the `cargo make test` target:

```bash
[cargo-make][1] INFO - Project: span
[cargo-make][1] INFO - Build File: Makefile.toml
[cargo-make][1] INFO - Task: test
[cargo-make][1] INFO - Profile: development
[cargo-make][1] INFO - Running Task: test
[cargo-make][1] INFO - Execute Command: "cargo" "test"
   Compiling span v0.1.0 (/home/ckuehl/src/enarx/span)
    Finished test [unoptimized + debuginfo] target(s) in 0.35s
     Running target/debug/deps/span-e770ea83353494aa

running 8 tests
test line::test::convert ... ok
test line::test::contains ... ok
test line::test::is_empty ... ok
test line::test::split ... ok
test span::test::contains ... ok
test span::test::convert ... ok
test span::test::is_empty ... ok
test span::test::split ... ok

test result: ok. 8 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests span
error[E0659]: `span` is ambiguous (name vs any other name during import resolution)
  --> /home/ckuehl/src/enarx/span/src/lib.rs:18:9
   |
18 | pub use span::Span;
   |         ^^^^ ambiguous name
   |
   = note: `span` could refer to a crate passed with `--extern`
   = help: use `::span` to refer to this crate unambiguously
note: `span` could also refer to the module defined here
  --> /home/ckuehl/src/enarx/span/src/lib.rs:15:1
   |
15 | mod span;
   | ^^^^^^^^^
   = help: use `crate::span` to refer to this module unambiguously

error: aborting due to previous error

For more information about this error, try `rustc --explain E0659`.
error: test failed, to rerun pass '--doc'
[cargo-make][1] ERROR - Error while executing command, exit code: 1
[cargo-make][1] WARN - Build Failed.
[cargo-make] ERROR - Error while executing command, exit code: 1
[cargo-make] WARN - Build Failed.
[ckuehl@redhat enarx]$ rustc --version
rustc 1.43.0 (4fb7144ed 2020-04-20)
```

Reply:
I wonder if that is a `1.43.0`ism. It appears to be ambiguous on the crate vs the module. I think I'd prefer to rename the crate.